Reference doc null, createObject and coalesce (#1011)
PowerShell/DSC: docs/reference/schemas/config/functions/createObject.md

  result:
    actualState:
      output:
        simpleObject:
          name: test
        multipleProps:
          key1: value1
          key2: 42
        emptyObject: {}
messages: []
hadErrors: false
```

### Example 2 - Mixed data types

The following example shows how to create objects with different value types.

```yaml
# createObject.example.2.dsc.config.yaml
$schema: https://aka.ms/dsc/schemas/v3/bundled/config/document.json
resources:
- name: Mixed data types
  type: Microsoft.DSC.Debug/Echo
  properties:
    output: "[createObject('string', 'hello', 'number', 123, 'boolean', true, 'nullValue', null())]"
```

```bash
dsc config get --file createObject.example.2.dsc.config.yaml
```

```yaml
results:
- name: Mixed data types
  type: Microsoft.DSC.Debug/Echo
  result:
    actualState:
      output:
        string: hello
        number: 123
        boolean: true
        nullValue: null
messages: []
hadErrors: false
```

### Example 3 - Nested objects and arrays

The following example shows how to create objects containing other objects and arrays.

```yaml
# createObject.example.3.dsc.config.yaml
$schema: https://aka.ms/dsc/schemas/v3/bundled/config/document.json
resources:

> - name: Nested structures
>   type: Microsoft.DSC.Debug/Echo
>   properties:
>     output:
>       nestedObject:   "[createObject('config', createObject('timeout', 30, 'enabled', true))]"
>       objectWithArray: "[createObject('items', createArray('Yoda', 'Darth Vader', 'Palpetine'))]"

Reviewer: ```suggestion
- name: Nested structures
  type: Microsoft.DSC.Debug/Echo
  properties:
    output:
      nestedObject:   "[createObject('config', createObject('timeout', 30, 'enabled', true))]"
      objectWithArray: "[createObject('items', createArray('foo', 'bar', 'baz'))]"
```

Author: Gehe, it was worth a try.

```

```bash
dsc config get --file createObject.example.3.dsc.config.yaml
```

```yaml
results:
- name: Nested structures
  type: Microsoft.DSC.Debug/Echo
  result:
    actualState:
      output:
        nestedObject:
          config:
            timeout: 30
            enabled: true
        objectWithArray:
          items:

>           - Yoda
>           - 'Darth Vader'
>           - Palpetine

Reviewer: ```suggestion
          - foo
          - bar
          - baz
```

messages: []
hadErrors: false
```

### Example 4 - Using with other functions

The following example shows how to use `createObject()` with other DSC functions.

```yaml
# createObject.example.4.dsc.config.yaml
$schema: https://aka.ms/dsc/schemas/v3/bundled/config/document.json
parameters:
  userName:
    type: string
    defaultValue: guest
resources:
- name: Function integration
  type: Microsoft.DSC.Debug/Echo
  properties:
    output:
      userConfig:     "[createObject('user', parameters('userName'), 'role', coalesce(null(), 'default'))]"
      fallbackObject: "[createObject('result', coalesce(null(), createObject('status', 'success')))]"
```

```bash
dsc config get --file createObject.example.4.dsc.config.yaml
```

```yaml
results:
- name: Function integration
  type: Microsoft.DSC.Debug/Echo
  result:
    actualState:
      output:
        userConfig:
          user: guest
          role: default
        fallbackObject:
          result:
            status: success
messages: []
hadErrors: false
```

## Parameters

> ### key, value pairs

Reviewer: ```suggestion
### Key-value pairs
```

The `createObject()` function accepts zero or more key-value pairs. Each key must be a string,
and values can be of any type.

```yaml
Type:         key: [string], value: [any]
Required:     false
MinimumCount: 0
MaximumCount: unlimited (must be even number)
```

#### key

The object property name. Must be a string value.

```yaml
Type:     [string]
Required: true (when providing values)
```

#### value

The object property value. Can be any valid JSON type including strings, numbers, booleans, null, arrays, or other objects.

```yaml
Type:     [any]
Required: true (when providing keys)
```

## Output

The `createObject()` function returns a JSON object containing the specified key-value pairs.

```yaml
Type: [object]
```

## Error conditions

The function will return an error in the following cases:

- **Odd number of arguments**: Arguments must be provided in key-value pairs
- **Non-string keys**: All keys must be string values
- **Invalid argument types**: Arguments must be valid JSON types

## Notes

> - Keys must be strings; numeric or other types are not allowed as keys
> - Values can be any valid JSON type including null, arrays, and nested objects
> - Duplicate keys will result in the last value overwriting previous values
> - Empty object creation (`createObject()` with no arguments) is supported
> - The function preserves the order of properties as specified

Reviewer: ```suggestion
- Keys must be strings. If you specify numeric or other types, DSC raises an error.
- Values can be any valid JSON type including null, arrays, and nested objects.
- Duplicate keys will result in the last value overwriting previous values.
- Empty object creation (`createObject()` with no arguments) is supported.
- The function preserves the order of properties as specified.
```
